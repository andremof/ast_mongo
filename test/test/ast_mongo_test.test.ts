import * as DEBUG from 'debug';
import * as path from 'path';
import {
    AstMongo, AstMongoOptions, StaticProperties,
    StaticModelHelper, StaticModelHelperCB,
    EndpointSet, EndpointSetHelper, EndpointSetHelperCB
} from 'ast_mongo_ts';
import { AstUtils, AstUtilsConfg } from 'ast_utils';
import { Pjsua,
PjsuaConfigs,
PlayerConfig,
AccountConfig,
AccountExt,
CallExt,
CallInfo,
} from 'pjsua';

const debug = DEBUG('AST_MONGO:tester');

const ENV = process.env;
const Domain = 'asterisk';
const MyID = '6001';
const BadID = 'nouser';
const CalleeLongID = '300';
const CalleeShortID = '301';
const HostAddress = ENV.ASTERISK_ADDRESS || '127.0.0.1';
const MyURI = `sip:${MyID}@${Domain}`;
const BadURI = `sip:${BadID}@${Domain}`;
const CalleeLongURI = `sip:${CalleeLongID}@${HostAddress}`;
const CalleeShortURI = `sip:${CalleeShortID}@${HostAddress}`;
const RegistrarURI = `sip:${HostAddress}:5060`;
const AmiUser = 'asterisk';
const AmiPassword = 'asterisk';
const ShortSound = 'demo-thanks';
const LongSound = 'demo-instruct';
const AstMakeAShortCall = `channel originate PJSIP/${MyID} application playback ${ShortSound}`;
const AstMakeALongCall = `channel originate PJSIP/${MyID} application playback ${LongSound}`;

const LOG_LEVEL = ENV.LOG_LEVEL || ENV.npm_package_config_pjsua_loglevel || 3;
const astMongoOptions: AstMongoOptions = {
    urls: {
        config: ENV.MONGO_CONFIG || ENV.npm_package_config_config || 'mongodb://127.0.0.1:27017/config_test',
        cdr: ENV.MONGO_CDR || ENV.npm_package_config_cdr || 'mongodb://127.0.0.1:27017/cdr_test',
        cel: ENV.MONGO_CEL || ENV.npm_package_config_cel || 'mongodb://127.0.0.1:27017/cel_test'
    },
};

const astUtilsConfig: AstUtilsConfg = {
    host: HostAddress,
    ari: {
        protocol: 'http',
        port: 8088,
        username: AmiUser,
        password: AmiPassword
    },
    ami: {
        port: 5038,
        username: AmiUser,
        password: AmiPassword
    }
};

const unique_id = '000000000000000000000001';

let ua: Pjsua;
let ast_mongo: AstMongo;
let ast_utils: AstUtils;
let smh: StaticModelHelper;
let esh: EndpointSetHelper;

class Timeout {
    private static readonly DEFAULT_TIMEOUT = 20 * 1000; // msec
    private readonly _saved: number;
    constructor(timeout = Timeout.DEFAULT_TIMEOUT) {
        this._saved = jasmine.DEFAULT_TIMEOUT_INTERVAL;
        jasmine.DEFAULT_TIMEOUT_INTERVAL = timeout;
    }
    end(): void {
        jasmine.DEFAULT_TIMEOUT_INTERVAL = this._saved;
    }
}

function delay(sec: number): Promise<void> {
    return new Promise(resolve => setTimeout(resolve, sec * 1000));
}

/**
 * Trim the result and adjust it for differences with older and newer asterisk
 */
function adjustOutput(result: object): string {
    const head = /^\s+/gm;
    const tail = /\s+$/gm;
    const end = /^--END COMMAND/gm;
    const pbx_config = /\[pbx_config\]/gm;
    let output = result['$content'] || result.Output;
    if (!output)
        return '';
    output = output
        .replace(head, '')
        .replace(tail, '')
        .replace(end, '')
        .replace(pbx_config, '[extensions.conf:0]');
    if (output.slice(-1) !== '\n')
        output = output + '\n';
    return output;
}

beforeAll( async () => {

    global.Promise = Promise;
    jest.setTimeout(60 * 1000);

    debug('connecting AstMongo...');
    ast_mongo = new AstMongo(astMongoOptions);
    await ast_mongo.connect();

    debug('connecting AstUtils...');
    ast_utils = new AstUtils(astUtilsConfig);
    await ast_utils.connect();

    smh = new StaticModelHelper(ast_mongo.Static, unique_id);
    esh = new EndpointSetHelper(ast_mongo);

    await ast_mongo.Aor.remove({});
    await ast_mongo.Auth.remove({});
    await ast_mongo.Endpoint.remove({});
    await ast_mongo.Cdr.remove({});
    await ast_mongo.Cel.remove({});

    debug('preparing pjsip.conf...');
    const pjsip = await smh.create(
        'pjsip.conf', 'transport-udp', [{
            type: 'transport',
            protocol: 'udp',
            bind: '0.0.0.0'
        }]);
    await ast_mongo.Static.create(pjsip);
    await ast_utils.reload(10 * 1000);

    debug('preparing extensions.conf...');
    const extensions = await smh.create(
        'extensions.conf', 'default', [
            // { exten: '300,1,Playback(demo-instruct)'},
            // { exten: '300,n,Hangup()'},
            // { exten: '301,1,Playback(demo-instruct)'},
            // { exten: '301,n,Hangup()'},
            { exten: '_6.,1,NoOp()'},
            { exten: '_6.,n,Dial(PJSIP/${EXTEN})'},
            { exten: '_6.,n,Hangup()'},
        ]);
    await ast_mongo.Static.create(extensions);
    await ast_utils.reloadDialPlan();

    debug('create an endpoint');
    await esh.create(MyID, {
        endpoint: {
            context: 'default',
            allow: 'ulaw',
            transport: 'transport-udp',
            rewrite_contact: true
        },
        aor: {
            max_contacts: 1,
            remove_existing: true,
            contact: `sip:${MyID}@tester:5060`
        },
        auth: {
            auth_type: 'userpass',
            password: MyID,
            username: MyID
        }
    });
});

afterAll(() => {
    ast_utils.disconnect();
});

describe('ast_mongo', () => {

    test ('check pjsip resources of the remote asterisk', async () => {
        // 1
        const dialplan = await ast_utils.exec('dialplan show');
        debug('check pjsip resources of the remote asterisk - dialplan', dialplan);
        expect(adjustOutput(dialplan)).toMatchSnapshot();
        // 2
        const transports = await ast_utils.exec('pjsip show transports');
        debug('check pjsip resources of the remote asterisk - transports', transports);
        expect(adjustOutput(transports)).toMatchSnapshot();
        // 3
        const transport = await ast_utils.exec('pjsip show transport transport-udp');
        debug('check pjsip resources of the remote asterisk - transport', transport);
        expect(adjustOutput(transport)).toMatchSnapshot();
        // 4
        const endpoints = await ast_utils.exec('pjsip show endpoints');
        debug('check pjsip resources of the remote asterisk - endpoints', endpoints);
        // modify to compare with the snap shot
        endpoints.Output = adjustOutput(endpoints)
            .replace(/(Not.in.use.|Invalid....)/g, "Unavailable")
            .replace(/Contact:  6001\/sip:6001@172(.*)(Unknown|NonQual)(.*)\n/gm, "")
            .replace(/Contact:  6001\/sip:6001@tester:5060                  9d7987fe43 (Unknown|NonQual)         nan$/gm,
                      "Contact:  6001/sip:6001@tester:5060                  9d7987fe43 Created       0.000");
        expect(endpoints.Output).toMatchSnapshot();
        // 5
        const endpoint = await ast_utils.exec(`pjsip show endpoint ${MyID}`);
        debug('check pjsip resources of the remote asterisk - endpoint', endpoint);
        // modify to compare with the snap shot
        endpoint.Output = adjustOutput(endpoint)
            .replace(/(Not.in.use.|Invalid....)/g, "Unavailable")
            .replace(/Contact:  6001\/sip:6001@172(.*)\n/gm, "")
            .replace(/Contact:  6001\/sip:6001@tester:5060                  9d7987fe43 (Unknown|NonQual)         nan$/gm,
                      "Contact:  6001/sip:6001@tester:5060                  9d7987fe43 Created       0.000")
            .replace(/accept_multiple_sdp_answers        : false\n/g, "")
            .replace(/follow_early_media_fork            : true\n/g, "")
            .replace(/mwi_subscribe_replaces_unsolicited : (false|no)\n/g, "")
            .replace(/suppress_q850_reason_headers       : false\n/g, "");
        expect(endpoint.Output).toMatchSnapshot();
        // 6
        const aors = await ast_utils.exec('pjsip show aors');
        debug('check pjsip resources of the remote asterisk - aors', aors);
        // modify to compare with the snap shot
        aors.Output = adjustOutput(aors)
            .replace(/^Contact:  6001\/sip:6001@172(.*)\n/gm, "")
            .replace(/NonQual/g, "Unknown");
        expect(aors.Output).toMatchSnapshot();
        // 7
        const aor = await ast_utils.exec(`pjsip show aor ${MyID}`);
        debug('check pjsip resources of the remote asterisk - aor', aor);
        // modify to compare with the snap shot
        aor.Output = adjustOutput(aor)
            .replace(/^=+\n/gm, "")
            .replace(/Contact:  6001\/sip:6001@172(.*)\n/gm, "")
            .replace(/contact              : sip:6001@172(.*)\n/gm, "")
            .replace(/NonQual/g, "Unknown");
        expect(aor.Output).toMatchSnapshot();
        // 8
        const auths = await ast_utils.exec('pjsip show auths');
        debug('check pjsip resources of the remote asterisk - auths', auths);
        expect(adjustOutput(auths)).toMatchSnapshot();
        // 9
        const auth = await ast_utils.exec(`pjsip show auth ${MyID}`);
        debug('check pjsip resources of the remote asterisk - auth', auth);
        expect(adjustOutput(auth)).toMatchSnapshot();
    });

    test ('create a pjsua', done => {
        debug('creating a ua', __dirname);
        const config: PjsuaConfigs = {
            endpoint: {
                logConfig: {
                    level: LOG_LEVEL,
                    consoleLevel: LOG_LEVEL
                }
            },
            transport: {
                port: 5060,
            },
            player: {
                player: {
                    filename: path.join(__dirname, '../waves', 'sound.wav')
                },
                recorder: {
                    filename: path.join(__dirname, '../waves', 'call.wav')
                }
            }
        };
        ua = new Pjsua(config);
        done();
    });

    test ('make an account', async () => {
        debug('making an account');
        const config: AccountConfig = {
            idUri: MyURI,
            regConfig: {
                registrarUri: RegistrarURI,
            },
            sipConfig: {
                authCreds: [{
                    scheme: 'digest',
                    realm: Domain,
                    username: MyID,
                    dataType: 0,
                    data: MyID
                }],
            }
        };
        await ua.makeAccount(config);
    });

    test ('wait for a call, accept it, then disconnect from far', async (done) => {
        ua.account.once('call', async (info: CallInfo, call: CallExt) => {
            debug('account.onCall & accept it');
            await ua.account.answer(call);
            ua.account.call.once('disconnected', async () => {
                debug('ua.account.call.disconnected');
                await delay(1); // to check cdr and cel
                done();
            });
        });
        debug('make an inbound call');
        const result = await ast_utils.exec(AstMakeAShortCall);
        debug('wait for a call, accept it, then disconnect from far - result', result);
        expect(adjustOutput(result)).toMatchSnapshot();
    });

    test ('check cdr', async () => {
        const results = await ast_mongo.Cdr
            .find()
            .sort('start')
            .select('-_id -start -answer -end -linkedid -uniqueid');
        // modify to compare with the snap shot
        const objs = JSON.parse(JSON.stringify(results));
        objs.forEach(result => {
            if (result.channel)
                result.channel = result.channel.split('-')[0];
            if (result.sequence && typeof result.sequence === 'number')
                result.sequence = 0;
        });
        expect(objs).toMatchSnapshot();
    });

    test ('check cel', async () => {
        const results = await ast_mongo.Cel
            .find()
            .sort('eventtime')
            .select('-_id -eventtime -linkedid -uniqueid');
        // modify to compare with the snap shot
        const objs = JSON.parse(JSON.stringify(results));
        objs.forEach(result => {
            if (result.extra)
                result.extra = JSON.parse(result.extra);
            if (result.channame)
                result.channame = result.channame.split('-')[0];
        });
        expect(objs).toMatchSnapshot();
    });
});
