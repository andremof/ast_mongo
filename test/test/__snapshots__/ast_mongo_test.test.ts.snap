// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ast_mongo check cdr 1`] = `
Array [
  Object {
    "accountcode": "",
    "amaflags": "DOCUMENTATION",
    "billsec": 5,
    "channel": "PJSIP/6001",
    "clid": "\\"\\" <>",
    "dcontext": "default",
    "disposition": "ANSWERED",
    "dst": "s",
    "dstchannel": "",
    "duration": 5,
    "lastapp": "Playback",
    "lastdata": "demo-thanks",
    "peeraccount": "",
    "sequence": 0,
    "src": "",
    "userfield": "",
  },
]
`;

exports[`ast_mongo check cel 1`] = `
Array [
  Object {
    "accountcode": "",
    "appdata": "",
    "appname": "",
    "channame": "PJSIP/6001",
    "cid_ani": "",
    "cid_dnid": "",
    "cid_name": "",
    "cid_num": "",
    "cid_rdnis": "",
    "context": "default",
    "eventname": "CHAN_START",
    "eventtype": 1,
    "exten": "s",
    "extra": "",
    "peer": "",
    "peeraccount": "",
    "userfield": "",
  },
  Object {
    "accountcode": "",
    "appdata": "(Outgoing Line)",
    "appname": "AppDial2",
    "channame": "PJSIP/6001",
    "cid_ani": "",
    "cid_dnid": "",
    "cid_name": "",
    "cid_num": "",
    "cid_rdnis": "",
    "context": "default",
    "eventname": "ANSWER",
    "eventtype": 4,
    "exten": "s",
    "extra": "",
    "peer": "",
    "peeraccount": "",
    "userfield": "",
  },
  Object {
    "accountcode": "",
    "appdata": "(Outgoing Line)",
    "appname": "AppDial2",
    "channame": "PJSIP/6001",
    "cid_ani": "",
    "cid_dnid": "",
    "cid_name": "",
    "cid_num": "",
    "cid_rdnis": "",
    "context": "default",
    "eventname": "HANGUP",
    "eventtype": 3,
    "exten": "s",
    "extra": Object {
      "dialstatus": "",
      "hangupcause": 0,
      "hangupsource": "",
    },
    "peer": "",
    "peeraccount": "",
    "userfield": "",
  },
  Object {
    "accountcode": "",
    "appdata": "(Outgoing Line)",
    "appname": "AppDial2",
    "channame": "PJSIP/6001",
    "cid_ani": "",
    "cid_dnid": "",
    "cid_name": "",
    "cid_num": "",
    "cid_rdnis": "",
    "context": "default",
    "eventname": "CHAN_END",
    "eventtype": 2,
    "exten": "s",
    "extra": "",
    "peer": "",
    "peeraccount": "",
    "userfield": "",
  },
]
`;

exports[`ast_mongo check pjsip resources of the remote asterisk 1`] = `
Object {
  "Message": "Command output follows",
  "Output": "[ Context '__func_periodic_hook_context__' created by 'func_periodic_hook' ]
'beep' (CID match '') =>  1. Answer()                                   [func_periodic_hook]
2. Playback(beep)                             [func_periodic_hook]
'hook' (CID match '') =>  1. Set(EncodedChannel=\${CUT(HOOK_CHANNEL,-,1-2)}) [func_periodic_hook]
2. Set(GROUP_NAME=\${EncodedChannel}\${HOOK_ID}) [func_periodic_hook]
3. Set(GROUP(periodic-hook)=\${GROUP_NAME})    [func_periodic_hook]
4. ExecIf($[\${GROUP_COUNT(\${GROUP_NAME}@periodic-hook)} > 1]?Hangup()) [func_periodic_hook]
5. Set(ChannelToSpy=\${URIDECODE(\${EncodedChannel})}) [func_periodic_hook]
6. ChanSpy(\${ChannelToSpy},qEB)               [func_periodic_hook]

[ Context 'default' created by 'pbx_config' ]
'_6.' =>          1. NoOp()                                     [extensions.conf:0]
2. Dial(PJSIP/\${EXTEN})                       [extensions.conf:0]
3. Hangup()                                   [extensions.conf:0]

-= 3 extensions (11 priorities) in 2 contexts. =-",
  "Response": "Success",
}
`;

exports[`ast_mongo check pjsip resources of the remote asterisk 2`] = `
Object {
  "Message": "Command output follows",
  "Output": "Transport:  <TransportId........>  <Type>  <cos>  <tos>  <BindAddress....................>
==========================================================================================

Transport:  transport-udp             udp      0      0  0.0.0.0:5060

Objects found: 1
",
  "Response": "Success",
}
`;

exports[`ast_mongo check pjsip resources of the remote asterisk 3`] = `
Object {
  "Message": "Command output follows",
  "Output": "Transport:  <TransportId........>  <Type>  <cos>  <tos>  <BindAddress....................>
==========================================================================================

Transport:  transport-udp             udp      0      0  0.0.0.0:5060

ParameterName              : ParameterValue
===========================================
allow_reload               : false
async_operations           : 1
bind                       : 0.0.0.0:5060
ca_list_file               :
ca_list_path               :
cert_file                  :
cipher                     :
cos                        : 0
domain                     :
external_media_address     :
external_signaling_address :
external_signaling_port    : 0
local_net                  :
method                     : unspecified
password                   :
priv_key_file              :
protocol                   : udp
require_client_cert        : No
symmetric_transport        : false
tos                        : 0
verify_client              : No
verify_server              : No
websocket_write_timeout    : 100
",
  "Response": "Success",
}
`;

exports[`ast_mongo check pjsip resources of the remote asterisk 4`] = `
Object {
  "Message": "Command output follows",
  "Output": "Endpoint:  <Endpoint/CID.....................................>  <State.....>  <Channels.>
I/OAuth:  <AuthId/UserName...........................................................>
Aor:  <Aor............................................>  <MaxContact>
Contact:  <Aor/ContactUri..........................> <Hash....> <Status> <RTT(ms)..>
Transport:  <TransportId........>  <Type>  <cos>  <tos>  <BindAddress..................>
Identify:  <Identify/Endpoint.........................................................>
Match:  <criteria.........................>
Channel:  <ChannelId......................................>  <State.....>  <Time.....>
Exten: <DialedExten...........>  CLCID: <ConnectedLineCID.......>
==========================================================================================

Endpoint:  6001                                                 Unavailable   0 of inf
InAuth:  6001/6001
Aor:  6001                                               1
Contact:  6001/sip:6001@tester:5060                  9d7987fe43 Created       0.000
Transport:  transport-udp             udp      0      0  0.0.0.0:5060


Objects found: 1
",
  "Response": "Success",
}
`;

exports[`ast_mongo check pjsip resources of the remote asterisk 5`] = `
Object {
  "Message": "Command output follows",
  "Output": "Endpoint:  <Endpoint/CID.....................................>  <State.....>  <Channels.>
I/OAuth:  <AuthId/UserName...........................................................>
Aor:  <Aor............................................>  <MaxContact>
Contact:  <Aor/ContactUri..........................> <Hash....> <Status> <RTT(ms)..>
Transport:  <TransportId........>  <Type>  <cos>  <tos>  <BindAddress..................>
Identify:  <Identify/Endpoint.........................................................>
Match:  <criteria.........................>
Channel:  <ChannelId......................................>  <State.....>  <Time.....>
Exten: <DialedExten...........>  CLCID: <ConnectedLineCID.......>
==========================================================================================

Endpoint:  6001                                                 Unavailable   0 of inf
InAuth:  6001/6001
Aor:  6001                                               1
Contact:  6001/sip:6001@tester:5060                  9d7987fe43 Created       0.000
Transport:  transport-udp             udp      0      0  0.0.0.0:5060


ParameterName                      : ParameterValue
=========================================================
100rel                             : yes
accountcode                        :
acl                                :
aggregate_mwi                      : true
allow                              : (ulaw)
allow_overlap                      : true
allow_subscribe                    : true
allow_transfer                     : true
aors                               : 6001
asymmetric_rtp_codec               : false
auth                               : 6001
bind_rtp_to_media_address          : false
bundle                             : false
call_group                         :
callerid                           : <unknown>
callerid_privacy                   : allowed_not_screened
callerid_tag                       :
connected_line_method              : invite
contact_acl                        :
context                            : default
cos_audio                          : 0
cos_video                          : 0
device_state_busy_at               : 0
direct_media                       : true
direct_media_glare_mitigation      : none
direct_media_method                : invite
disable_direct_media_on_nat        : false
dtls_auto_generate_cert            : No
dtls_ca_file                       :
dtls_ca_path                       :
dtls_cert_file                     :
dtls_cipher                        :
dtls_fingerprint                   : SHA-256
dtls_private_key                   :
dtls_rekey                         : 0
dtls_setup                         : active
dtls_verify                        : No
dtmf_mode                          : rfc4733
fax_detect                         : false
fax_detect_timeout                 : 0
force_avp                          : false
force_rport                        : true
from_domain                        :
from_user                          :
g726_non_standard                  : false
ice_support                        : false
identify_by                        : username,ip
inband_progress                    : false
incoming_mwi_mailbox               :
language                           :
mailboxes                          :
max_audio_streams                  : 1
max_video_streams                  : 1
media_address                      :
media_encryption                   : no
media_encryption_optimistic        : false
media_use_received_transport       : false
message_context                    :
moh_passthrough                    : false
moh_suggest                        : default
mwi_from_user                      :
named_call_group                   :
named_pickup_group                 :
notify_early_inuse_ringing         : false
one_touch_recording                : false
outbound_auth                      :
outbound_proxy                     :
pickup_group                       :
preferred_codec_only               : false
record_off_feature                 : automixmon
record_on_feature                  : automixmon
refer_blind_progress               : true
rewrite_contact                    : true
rpid_immediate                     : false
rtcp_mux                           : false
rtp_engine                         : asterisk
rtp_ipv6                           : false
rtp_keepalive                      : 0
rtp_symmetric                      : false
rtp_timeout                        : 0
rtp_timeout_hold                   : 0
sdp_owner                          : -
sdp_session                        : Asterisk
send_diversion                     : true
send_pai                           : false
send_rpid                          : false
set_var                            :
srtp_tag_32                        : false
sub_min_expiry                     : 0
subscribe_context                  :
t38_udptl                          : false
t38_udptl_ec                       : none
t38_udptl_ipv6                     : false
t38_udptl_maxdatagram              : 0
t38_udptl_nat                      : false
timers                             : yes
timers_min_se                      : 90
timers_sess_expires                : 1800
tone_zone                          :
tos_audio                          : 0
tos_video                          : 0
transport                          : transport-udp
trust_id_inbound                   : false
trust_id_outbound                  : false
use_avpf                           : false
use_ptime                          : false
user_eq_phone                      : false
voicemail_extension                :
webrtc                             : no
",
  "Response": "Success",
}
`;

exports[`ast_mongo check pjsip resources of the remote asterisk 6`] = `
Object {
  "Message": "Command output follows",
  "Output": "Aor:  <Aor..............................................>  <MaxContact>
Contact:  <Aor/ContactUri............................> <Hash....> <Status> <RTT(ms)..>
==========================================================================================

Aor:  6001                                                 1
Contact:  6001/sip:6001@tester:5060                    9d7987fe43 Unknown         nan


Objects found: 1
",
  "Response": "Success",
}
`;

exports[`ast_mongo check pjsip resources of the remote asterisk 7`] = `
Object {
  "Message": "Command output follows",
  "Output": "Aor:  <Aor..............................................>  <MaxContact>
Contact:  <Aor/ContactUri............................> <Hash....> <Status> <RTT(ms)..>

Aor:  6001                                                 1
Contact:  6001/sip:6001@tester:5060                    9d7987fe43 Unknown         nan


ParameterName        : ParameterValue
authenticate_qualify : false
contact              : sip:6001@tester:5060
default_expiration   : 3600
mailboxes            :
max_contacts         : 1
maximum_expiration   : 7200
minimum_expiration   : 60
outbound_proxy       :
qualify_frequency    : 0
qualify_timeout      : 3.000000
remove_existing      : true
support_path         : false
voicemail_extension  :
",
  "Response": "Success",
}
`;

exports[`ast_mongo check pjsip resources of the remote asterisk 8`] = `
Object {
  "Message": "Command output follows",
  "Output": "I/OAuth:  <AuthId/UserName.............................................................>
==========================================================================================

Auth:  6001/6001

Objects found: 1
",
  "Response": "Success",
}
`;

exports[`ast_mongo check pjsip resources of the remote asterisk 9`] = `
Object {
  "Message": "Command output follows",
  "Output": "I/OAuth:  <AuthId/UserName.............................................................>
==========================================================================================

Auth:  6001/6001

ParameterName  : ParameterValue
===============================
auth_type      : userpass
md5_cred       :
nonce_lifetime : 32
password       : 6001
realm          :
username       : 6001
",
  "Response": "Success",
}
`;

exports[`ast_mongo wait for a call, accept it, then disconnect from far 1`] = `
Object {
  "Message": "Command output follows",
  "Output": "",
  "Response": "Success",
}
`;
